internal/messaging/processor.go

```diff
@@ -141,14 +150,51 @@ func (p *processor) Request(ctx context.Context, msg *Message) (*Message, error)
 	if msg.Metadata.Recipient == "" { // TODO: add address validation
 		return nil, ErrMissingRecipient
 	}
-	msg.Metadata.Cheques = nil // TODO issue and attach cheques
+	p.logger.Infof("Distributor: received a request to propagate to CMAccount %s", msg.Metadata.Recipient)
+	cmAccountRecipient := msg.Metadata.Recipient
+	// lookup for CM Account -> bot
+	botAddress := CMAccountBotMap[common.HexToAddress(msg.Metadata.Recipient)]
```

Comment:
```suggestion
	botAddress, ok := CMAccountBotMap[common.HexToAddress(msg.Metadata.Recipient)]
```
and check for ok instead of botAddress == ""